### PR TITLE
[Merged by Bors] - feat: missing Nat.card lemmas

### PR DESCRIPTION
**Pull request by PatrickMassot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Cardinal/Finite.lean [hunk 1]
```diff
@@ -94,6 +94,21 @@ theorem card_eq_of_bijective (f : α → β) (hf : Function.Bijective f) : Nat.c
   card_congr (Equiv.ofBijective f hf)
 #align nat.card_eq_of_bijective Nat.card_eq_of_bijective
 
+protected theorem bijective_iff_injective_and_card [Fintype β] (f : α → β) :
```
Comment (apnelson1) [inline Mathlib/SetTheory/Cardinal/Finite.lean:97]: This should be true with `Nat.card β ≤ Nat.card α`, right? And it's probably is mostly used in the `.mpr` direction to prove bijectivity. Would the version with `≤` be more convenient?

Comment (PatrickMassot) [reply Mathlib/SetTheory/Cardinal/Finite.lean:97]: I really copied the Fintype version here, except that I removed the `[Fintype α]`. The issue with your version is that the conclusion is less strong in the `.mp` direction. So I think your version does not really make sense as an `iff`. Having the `.mpr` version with weaker assumption as a separate lemma would make sense.

Comment (PatrickMassot) [reply Mathlib/SetTheory/Cardinal/Finite.lean:97]: I added your lemma (only the meaningful implication).

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**